freqtrade/optimize/hyperopt.py

```diff
@@ -133,22 +136,44 @@ def log_trials_result(self) -> None:
         results = sorted(self.trials, key=itemgetter('loss'))
         best_result = results[0]
         params = best_result['params']
-
         log_str = self.format_results_logstring(best_result)
         print(f"\nBest result:\n\n{log_str}\n")
-        if self.has_space('buy'):
-            print('Buy hyperspace params:')
-            pprint({p.name: params.get(p.name) for p in self.hyperopt_space('buy')},
-                   indent=4)
-        if self.has_space('sell'):
-            print('Sell hyperspace params:')
-            pprint({p.name: params.get(p.name) for p in self.hyperopt_space('sell')},
-                   indent=4)
-        if self.has_space('roi'):
-            print("ROI table:")
-            pprint(self.custom_hyperopt.generate_roi_table(params), indent=4)
-        if self.has_space('stoploss'):
-            print(f"Stoploss: {params.get('stoploss')}")
+
+        if self.config.get('print_json'):
```

Comment:
For the future: 
this should probably be extracted to a function - so we can call it with "any" result - but i guess that's going to be part of the `print-result -n` PR